numpy/core/src/multiarray/datetime.c

```diff
@@ -722,9 +722,15 @@ parse_datetime_extended_unit_from_string(char const *str, Py_ssize_t len,
 {
     char const *substr = str, *substrend = NULL;
     int den = 1;
+    long long int true_meta_num;
```

Comment:
I may need `int64_t` or similar here to force a fixed-width integer of the size I want

Reply:
Hmmpf,
```
typedef struct {
    NPY_DATETIMEUNIT base;
    int num;
} PyArray_DatetimeMetaData;
```
defines it as an `int`, unfortunately, that is defined in the public header.

I don't see how changing the type will help, but maybe using `NumPyOS_strtoll` will.  I guess that function is also a bit wrong: It sets `errno` on overflow, but we never check for that, for example in `np.fromstring`…  I would no be surprised if it is easier to go via `PyLong_FromString `

Reply:
On platforms that treat `long int` as 64 bit integer it helps because the C function `strtol` used to return the value here uses that type, so I can check the "true value" before the typecast to shorter integer.

Unfortunately, on platforms where that is not true, it won't matter what I set the temporary check value to if the overflow has already happened, which may be what you are getting at (I wasn't trying to change the type of the public value, just the `true_meta_num` temporary storage value I check for overflow).

I had considered `PyLong_FromString`, I may indeed have to do something like that..